### ISSUE TITLE
New `pickle` API

### ISSUE THREAD

**Issue by bigbes** [tarantool/tarantool#1206]
It'll be partial copy of [python pickle API](https://docs.python.org/3/library/struct.html):

It'll be located in `require('pickle').future` namespace.

``` lua
local struct = require('pickle').future

-- For packing of structure
-- <output> must be buffer or socket/fd
struct.pack(<fmt>, <args>, ...)
struct.pack_into(<output>, <output>, <args>, ...)

-- For unpacking of structure
-- <source> may be buffer/string
struct.unpack(<fmt>, <source>, <offset>)


-- Return iterator for unpacking
struct.iter_unpack(<fmt>, <source>)
```

Advantages:
- Stream API (support for buffer or socket/fio). *socket/fio using buffering.
- Struct object for precompiled packing/unpacking.
- packing/unpacking using only parts of arguments.

Also suggested renaming in 1.7 (`require('pickle').future` -> `require('struct')`)

**Comment by bigbes** [comment]
It depends on adding `:sysread` and `:syswrite` into `fio_object` for adding support of streaming API.

**Comment by rtsisyk** [comment]
xrow reader will be implemented in C, so I remove this ticket of of 1.6. scope.

**Comment by kostja** [comment]
Nobody cares -> won't fix.
